### DISCUSSION TITLE
Use 'column_sort'-column to compute custom features

### DISCUSSION THREAD

**Discussion by DerMilchschnittenmann** [blue-yonder/tsfresh#1056] [Q&A]
Hi, I have a rolled df I want compute custom features on. For further processing I want to keep the first and last date of each 'feature-row'. I thought I could create custom-feature-functions that compute first and last date. Unfortunatly I cannot access the date_column (column_sort) inside the custom function. 
Is there any way one can do that?

Here is my example-code:

```python
import random
import pandas as pd
from datetime import datetime

from tsfresh import extract_features
from tsfresh.feature_extraction import feature_calculators
from tsfresh.feature_extraction import MinimalFCParameters
from tsfresh.utilities.dataframe_functions import roll_time_series


@feature_calculators.set_property("input", "pd.Series")
@feature_calculators.set_property("fctype", "simple")
def first_date(x):
    return x.index[0]



def start():
    # df
    date_range = pd.date_range(start=datetime(2000, 1, 1), end=datetime(2023, 12, 31), freq='MS')
    data = pd.DataFrame({
        'date': date_range,
        'value': [random.randint(0, 200) for _ in range(len(date_range))]
    })

    data['id'] = 'ID'

    # roll df
    data_rolled = roll_time_series(df_or_dict=data,
                                   column_id='id',
                                   column_sort='date',
                                   rolling_direction=6,
                                   max_timeshift=24,
                                   min_timeshift=24)
    
    # feature sets
    settings = MinimalFCParameters()
    settings[first_date] = None
    # settings[last_date] = None


    # extract features
    features = extract_features(timeseries_container=data_rolled,
                                default_fc_parameters=settings,
                                column_id='id',
                                column_sort='date',
                                column_value='value')

    print(features)


    # DESIRED OUTPUT:
    features['first_date'] = [tup[1]-pd.Timedelta(days=730) for tup in features.index]
    features['last_date'] = [tup[1] for tup in features.index]


if __name__ == '__main__':
    start()
```

**Comment by DerMilchschnittenmann** [comment]
I am sorry for the formatting but I cannot fix it :/ 
Here is the code as a txt.file
[test.txt](https://github.com/blue-yonder/tsfresh/files/13548057/test.txt)

**Reply by nils-braun** [reply]
Hi @DerMilchschnittenmann!

Concerning the code formatting: you can use triple-` and a new line for formatting :)

Now to your question:
You could use a date time feature for this: https://tsfresh.readthedocs.io/en/latest/text/how_to_add_custom_feature.html#writing-your-own-time-based-feature-calculators

But please note that tsfresh generally assumes that the extracted features are floats (not dates). Feature selection for example does not work with a date. 

May I ask why you need to use a custom feature extractor and the full tsfresh "machinery" and not just get the first and last date yourself (like you did in your expected output basically)? What benefit does using extract features of tsfresh bring you?
